File: src/main/java/ch/grassl/day03/Day03.java
package ch.grassl.day03;

import ch.grassl.common.AdventOfCodeDay;
import ch.grassl.day03.impl.importer.ItemImporter;
import ch.grassl.day03.impl.model.Rucksack;
import ch.grassl.day03.impl.RucksackService;

import java.util.List;

public class Day03 extends AdventOfCodeDay {

    private static final String RESOURCE = "/day3.txt";

    public static final int TEAM_SIZE = 3;

    private final List<Rucksack> rucksacks = ItemImporter.getInstance().importData(RESOURCE);
    private final RucksackService service = RucksackService.of(rucksacks);

    @Override
    public void run() {
        result.add(service.findPriorityScore());
        result.add(service.findTeamPriorityScore());
        printResult();
    }
}

File: src/main/java/ch/grassl/day03/Day03.kt
package ch.grassl.day03

import ch.grassl.common.AdventOfCodeDay
import ch.grassl.day03.impl.RucksackService
import ch.grassl.day03.impl.importer.ItemImporter

class Day03 : AdventOfCodeDay() {

    companion object {
        const val TEAM_SIZE = 3
        private const val RESOURCE = "/day3.txt"
    }
    private val rucksacks = ItemImporter().importData(RESOURCE)
    private val service = RucksackService.of(rucksacks)

    override fun run() {
        result.add(service.findPriorityScore())
        result.add(service.findTeamPriorityScore())
        printResult()
    }
}